### PR TITLE
Preload CSS and aggressive font caching

### PR DESCRIPTION
**Pull request by xHomu**
A couple performance tweaks:

* By default, HTML rendering is paused until CSS stylesheets streamed in! We can fix this by preload CSS as a resource, 
* Currently, fonts are cached only a hour like everything else in the Public folder, so let's set the cache header for fonts folder to a year instead.

### CODE REVIEW

File: app/root.tsx [hunk 1]
```diff
@@ -55,8 +55,12 @@ export const links: LinksFunction = () => {
 		},
 		{ rel: 'manifest', href: '/site.webmanifest' },
 		{ rel: 'icon', href: '/favicon.ico' },
+		// Preload CSS as a resource to avoid render blocking
+		{ rel: 'preload', href: fontStylestylesheetUrl, as: 'style' },
 		{ rel: 'stylesheet', href: fontStylestylesheetUrl },
+		{ rel: 'preload', href: tailwindStylesheetUrl, as: 'style' },
 		{ rel: 'stylesheet', href: tailwindStylesheetUrl },
+		cssBundleHref ? { rel: 'preload', href: cssBundleHref, as: 'style' } : null,
```
Comment (kentcdodds) [inline app/root.tsx:63]: Really doesn't feel like this should be necessary, or if it should, I would expect it to be better if we moved all preload tags to before the stylesheets come in?

Could you show some before/after metrics?

**Review by kentcdodds — COMMENTED** [review]
Thanks! Just one thing.

**Comment by xHomu** [comment]
> Really doesn't feel like this should be necessary, or if it should, I would expect it to be better if we moved all preload tags to before the stylesheets come in?
>
> Could you show some before/after metrics?

It's a bit verbose, but that line solves this particular warning in Lighthouse:

image

Details on preloading CSS: https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/rel/preload

The underlying theory is a bit over my head, but I do see it generally bump Lighthouse Performance by a point, and sometimes help with CLS.


Note: moving the preloads up to the top makes a lot of sense, so I pushed a

**Review by kentcdodds — APPROVED** [review]
Luckily I don't expect people to change these style imports much (ever) so I'm fine with it. Otherwise I think I may not be as jazzed about this 😅

Thanks!

**Pull request merged**